nestd/__init__.py

```diff
@@ -108,5 +108,31 @@ def get_all_nested(fx, *context_vars):
 
     return output
 
+def nested_deep_search(fx,inner_name,**free_vars):
```

Comment:
I have a question, why do we need `inner_name` as a parameter when we are returning all the nested functions?

I was thinking of a function that returns a dictionary with nested dictionaries.

Reply:
This is also a really nice implementation that I didn't think of. Also, correct me if I am wrong. Will replace the regular search?

Reply:
Ohh sorry My bad I was thinking from the start of removing only one function to search But i have read the Issue again I am the one at fault :grimacing:.I will fix it asap
#### Also that would be good to replace the normal search with `recursive` as if any thought of edge cases it will return `None` or a valid function after searching all the depths

Reply:
### Hi @sansyrox  i have some Doubts I didn't Get the `dictionary with nested dictionaries`
### is it like dictionary with `key(function name):value(function object)`
![Image not found](https://drive.google.com/uc?export=view&id=14QZS6o1Zz8GngtpIadNJi1tyY69zpuLa)

###  But also if we go in the above way wont there be any other function with the same name deep inside the tree  `In this case it will replace the dictionary`
### or isnt another Tuple output isn't good?

![Image not found](https://drive.google.com/uc?export=view&id=1HJRS_mXq9OCNa8TbqcQyg-123wyynCx2)

#### I am quite Confused.

Reply:
@rohanailoni , how would tuples show the following case? 

```
def fx():
   def a():
      def b():
          def c():
```


> have some Doubts I didn't Get the dictionary with nested dictionaries
is it like dictionary with key(function name):value(function object)

Yep. It will be like `key: function_object | dict`

If we have two functions of the same name on the same nested level, then I think we should be showing the latest function. Or we can append a hash in case of a collision. What do you think?


e.g.

```
def fx():
   def a(): # first function
       ....
   def a(): # second function
       ....
```

We can follow one of the two approaches:

```
output_dict = {
"a": function_object # second function
}
```

Or,

```
output_dict = {
"a": function_object # first function,
"a1": function_object # second function
}
```

The benefit of first method is that it is easier but we won't get all the functions. 

The benefit of the second method is that we will get all data but might have collisions with the subsequent methods.

I prefer the first one. But what do you think?

Reply:
> @rohanailoni , how would tuples show the following case?



```python
def f():
     def a():
           def b():
                  def c():
```
#### The tuple representation be like`["a":<function:a>,"b":<function:b>,"c":<function:c>]`

But i get the idea if the return is in dictionary then the user will also have choice to get the function in O(1) time
```json
{
   "a":"<function:a>",
    "b":"<function:b>",
    "c":"<function:c>"
}
```
#### Thanks for Clarification